components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/internal/IdpMgtListenerServiceComponent.java

```diff
@@ -33,21 +33,30 @@
  */
 public class IdpMgtListenerServiceComponent {
 
-    private static List<IdentityProviderMgtLister> listners = new ArrayList<IdentityProviderMgtLister>();
+    private static IdpMgtListenerServiceComponent instance = new IdpMgtListenerServiceComponent();
 
-    public static void setIdentityProviderMgtListerService(
+    private List<IdentityProviderMgtLister> listners = new ArrayList<IdentityProviderMgtLister>();
```

Comment:
Was any there any particular reason why this was made singleton? ServiceComponents by convention are not instantiated in other parts of the code, so this is something we can drop.